### ISSUE TITLE
Indirect and renamed imports are false positive "Unused exports in namespaces"

### ISSUE THREAD

**Issue by phiresky** [webpro/knip#452]
This is on both 3.x (current) and 4.0 (canary 14).
I _think_ this is a regression but not sure. I don't have a minimal reproducible example but it's hopefully clear from this:

### re-exports

```ts
// fileA
export const afoo = "foo";
// fileB
export * as A from "fileA";
// file C
import { A as x } from "fileB";
console.log(x.afoo);
```

This leads to
```
Unused exports in namespaces (1)
afoo             unknown  fileA
```

where interestingly the type of export is detected as "unknown"
 
### renamed star imports

```ts
// fileA
export const afoo = "foo";
// fileB
import * as a from "fileA";

const arenamed = a;

console.log(arenamed.afoo);
```

same error as above

both of these cases work fine with findReferences in VSCode so I'd expect them to also work in knip?;

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v4.0.0-canary.15. See [Release 4.0.0-canary.15](https://github.com/webpro/knip/releases/tag/4.0.0-canary.15) for release notes._

**Comment by webpro** [comment]
> This is on both 3.x (current) and 4.0 (canary 14).

> I think this is a regression but not sure.

I think this all works fine in v3.

> I don't have a minimal reproducible example but it's hopefully clear from this

Next time, please create it. It's not only about being clear, it's also good to have an actual reproduction.

> where interestingly the type of export is detected as "unknown"

Knip has a few known types, such as `interface`, `type`, `class` and `function` I believe, but not all yet.

> both of these cases work fine with findReferences in VSCode so I'd expect them to also work in knip?;

That's a good assumption, and it _should_ work like that, but in v4 this is no longer necessary the case (also see https://knip.dev/blog/slim-down-to-speed-up#the-story-of-findreferences).

The first case is now fixed, I've missed this case when refactoring to rely less on `findReferences`, thanks a bunch for reporting it!

The second case points out exactly the flaw of the same refactor. I was kinda surprised how little false positives it gives in both small and large codebases, but this is definitely something on the radar that needs to be fixed. I'm not going to let it block the v4 release, though. (Implementation-wise what I'll need is a way to do `typeChecker.getAliasedSymbol` for non-types.)
